src/_premake_init.lua

```diff
@@ -1348,6 +1344,13 @@
 	end)
 
 
+	-- 31 January 2017
+
+	api.deprecateValue("flags", "C++11", 'Use `language "C++11"` instead', function(value) end, function(value) end)
+	api.deprecateValue("flags", "C++14", 'Use `language "C++14"` instead', function(value) end, function(value) end)
+	api.deprecateValue("flags", "C90",   'Use `language "C90"` instead',   function(value) end, function(value) end)
+	api.deprecateValue("flags", "C99",   'Use `language "C99"` instead',   function(value) end, function(value) end)
+	api.deprecateValue("flags", "C11",   'Use `language "C11"` instead',   function(value) end, function(value) end)
```

Comment:
Ah that's right, forgot about that part. We need the ability to set the language (or standard, or whatever) for each "dialect".

```
clanguage "C99"
cpplanguage "C++14"
```

We can still have the generic `language`, which is used if a more specific value hasn't been set. And of course this needs to extend to any supported language:

```
dlanguage "SafeD"
```

So yeah…there's that.